src/ops/checkout.rs

```diff
@@ -80,3 +81,76 @@ fn checkout_new_branch_prompt_update(
     state.run_cmd(term, &[], cmd)?;
     Ok(())
 }
+
+pub(crate) struct DeleteBranch;
+impl OpTrait for DeleteBranch {
+    fn get_action(&self, _target: Option<&TargetData>) -> Option<Action> {
+        Some(Rc::new(move |state: &mut State, term: &mut Term| {
+            let rev = state.prompt(
+                term,
+                &PromptParams {
+                    prompt: "Delete branch",
+                    create_default_value: Box::new(selected_branch),
+                    ..Default::default()
+                },
+            )?;
+
+            delete_branch(state, term, &rev)?;
+            Ok(())
+        }))
+    }
+
+    fn display(&self, _state: &State) -> String {
+        "Delete branch".into()
+    }
+}
+
+fn delete_branch(state: &mut State, term: &mut Term, branch: &str) -> Res<()> {
+    if branch.is_empty() {
+        return Err(Error::InvalidBranch);
+    }
+
+    let is_unmerged = {
+        let current_branch = crate::git::get_current_branch(&state.repo)?;
+        let current_branch_name = match current_branch.name() {
+            Ok(Some(name)) => name,
+            Ok(None) => return Err(Error::CantGetBranchName),
+            Err(e) => return Err(Error::CantGetBranch(e)),
+        };
+
+        if branch == current_branch_name {
+            return Err(Error::CantDeleteCurrentBranch);
+        }
+
+        let target_branch = crate::git::get_branch(&state.repo, branch)?;
+
+        crate::git::is_branch_unmerged(&state.repo, &target_branch).unwrap_or(false)
+    };
+
+    if is_unmerged {
+        let branch_to_delete = branch.to_string();
+
+        match state.confirm(term, "Branch is unmerged. Really delete?") {
```

Comment:
```suggestion
        match state.confirm(term, "Branch is unmerged. Really delete? (y or n)") {
```

Ugly, but...
Let's throw in a `(y or n)` to make it consistent with the other ones? I should've just taken the effort to bake it into `state.confirm`, perhaps later.